### ISSUE TITLE
Category raster resampling uses wrong averaging method

### ISSUE THREAD

**Issue by Chrismarsh** [Chrismarsh/mesher#21]
When regularizing category rasters like landcover, `-r average` is used, which results in poor outputs due to values being in between integer values

**Comment by Chrismarsh** [comment]
GDAL produces the following warning 
```
Warning 1: Input file land_cover/NA_NALCMS_2015_v2_land_cover_30m/NA_NALCMS_2015_v2_land_cover_30m.tif has a color table, which will likely lead to bad results when using a resampling method other than nearest neighbour or mode. Converting the dataset prior to 24/32 bit is advised.
```

**Comment by Chrismarsh** [comment]
Resolved in 2.0.7
